argilla/src/argilla/_api/_token.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+from threading import Lock
+from typing import Optional
+from huggingface_hub.utils._runtime import is_google_colab
+from argilla._constants import _DEFAULT_API_URL
+
+
+_IS_GOOGLE_COLAB_CHECKED = False
+_GOOGLE_COLAB_SECRET_LOCK = Lock()
+_GOOGLE_COLAB_SECRET: Optional[dict] = None
+
+
+def get_token() -> Optional[dict]:
+    """
+    Get token if present
+
+    Returns:
+        `dict` or `None`: The token, `None` if it doesn't exist.
+    """
+    return _get_token_from_google_colab() or _get_token_from_environment()
+
+
+def _get_token_from_environment() -> Optional[str]:
```

Comment:
Should this output type be a dict?
```suggestion
def _get_token_from_environment() -> Dict[str,str]:
```